Microsoft.Toolkit.Uwp.UI.Controls/RemoteDevicePicker/RemoteDevicePicker.cs

```diff
@@ -0,0 +1,358 @@
+        /// </summary>
+        public static readonly DependencyProperty ShowAdvancedFiltersProperty = DependencyProperty.Register(
+            nameof(ShowAdvancedFilters),
+            typeof(bool),
+            typeof(RemoteDevicePicker),
+            new PropertyMetadata(false));
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RemoteDevicePicker"/> class.
+        /// </summary>
+        public RemoteDevicePicker()
+        {
+            DefaultStyleKey = typeof(RemoteDevicePicker);
+            RemoteSystems = new ObservableCollection<RemoteSystem>();
+            PrimaryButtonClick += RemoteDevicePicker_PrimaryButtonClick;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RemoteDevicePicker"/> class with filters.
+        /// </summary>
+        public RemoteDevicePicker(RemoteSystemDiscoveryType remoteSystemDiscoveryType, RemoteSystemAuthorizationKind remoteSystemAuthorizationKind, RemoteSystemStatusType remoteSystemStatusType)
+        {
+            DefaultStyleKey = typeof(RemoteDevicePicker);
+            RemoteSystems = new ObservableCollection<RemoteSystem>();
+            PrimaryButtonClick += RemoteDevicePicker_PrimaryButtonClick;
+
+            _discoveryFilter = new RemoteSystemDiscoveryTypeFilter(remoteSystemDiscoveryType);
+            _authorizationKindFilter = new RemoteSystemAuthorizationKindFilter(remoteSystemAuthorizationKind);
+            _statusFilter = new RemoteSystemStatusTypeFilter(remoteSystemStatusType);
+        }
+
+        private void RemoteDevicePicker_PrimaryButtonClick(ContentDialog sender, ContentDialogButtonClickEventArgs args)
+        {
+            _listDevices.SelectedItems.Clear();
+        }
+
+        /// <summary>
+        /// Initiate Picker UI
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task<IEnumerable<RemoteSystem>> PickDeviceAsync()
+        {
+            await ShowAsync();
+            IEnumerable<RemoteSystem> devices = ReturnDevices();
+            return devices;
+        }
+
+        internal IEnumerable<RemoteSystem> ReturnDevices()
+        {
+            List<RemoteSystem> returnValue = new List<RemoteSystem>();
+            foreach (RemoteSystem item in _listDevices.SelectedItems)
+            {
+                returnValue.Add(item);
+            }
+
+            return returnValue.AsEnumerable();
+        }
+
+        /// <inheritdoc />
+        protected override void OnApplyTemplate()
+        {
+            UnhookEvents();
+
+            _listDevices = GetTemplateChild("PART_ListDevices") as ListView;
+            _listDeviceTypes = GetTemplateChild("PART_ListDeviceTypes") as ComboBox;
+            _progressRing = GetTemplateChild("PART_ProgressRing") as ProgressRing;
+            _commandSpace = GetTemplateChild("CommandSpace") as Grid;
```

Comment:
I feel like you don't need this _commandSpace Grid reference just for a visibility.  If someone retemplated without a ListView, they can control whether they show buttons or not.

Reply:
Surrounded this with null check that way it will not break for retemplating.